### PR TITLE
Simplify algebra for  x / 1 operations

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Done to satisfy simplifications specified by #1236 .  Just replace every  parameter divided by 1 with itself. It's assumed that the eliminate_identity() pass will handle generated identity operators in our run_pass()

### CODE REVIEW

**Review by umangyadav — APPROVED** [review]

**Comment by umangyadav** [comment]
@pfultz2  We are not running `simplify_algebra` after propagate_constant, I think we may miss some optimizations in doing so.

File: test/simplify_algebra_test.cpp [hunk 1]
```diff
@@ -734,6 +734,25 @@ TEST_CASE(simplify_div_const)
     EXPECT(m1 == m2);
 }
 
+TEST_CASE(simplify_unit_div_const)
+{
+    migraphx::module m1;
+    {
+        auto x    = m1.add_parameter("x", {migraphx::shape::int32_type, {1}});
+        auto unit = m1.add_literal(1);
+        m1.add_instruction(migraphx::make_op("div"), x, unit);
```
Comment (pfultz2) [inline test/simplify_algebra_test.cpp:743]: I think you can add a return or just add in an extra operator so it doesnt get replaced with identity:

```cpp
        auto unit = m1.add_literal(1);
        auto div = m1.add_instruction(migraphx::make_op("div"), x, unit);
        m1.add_instruction(migraphx::make_op("relu"), div);
```

```cpp
        auto unit = m1.add_literal(1);
        auto div = m1.add_instruction(migraphx::make_op("div"), x, unit);
        m1.add_return({div});
```

**Comment by umangyadav** [comment]
> > Should we move these 'identity' transformations in another pass and call it after propagate_constant ?
> 
> No. Ideally, I would like to make a super pass that has several optimizations in a loop like simplify_algebra, simplify_reshapes, CSE, propagate_constant, etc.

That is a better idea, but ideally, we should loop into that until the program stop changing. I am not sure given the transformations we have, it is guaranteed to converge, (it could be, but not sure). Otherwise, loop trip count becomes a tuning parameter that would affect performance.

**Comment by pfultz2** [comment]
> That is a better idea, but ideally, we should loop into that until the program stop changing.

Yea, we need a way to track changes to the module. Perhaps a signal? Or we could have finders return if they change the graph?

> Otherwise, loop trip count becomes a tuning parameter that would affect performance.

Well we already have loop counts already. Either way, we would want to have a max loop count , but we could have a compile_options flag to increase it.

**Comment by TedThemistokleous** [comment]
> > That is a better idea, but ideally, we should loop into that until the program stop changing.
> 
> Yea, we need a way to track changes to the module. Perhaps a signal? Or we could have finders return if they change the graph?
> 
> > Otherwise, loop trip count becomes a tuning parameter that would affect performance.
> 
> Well we already have loop counts already. Either way, we would want to have a max loop count , but we could have a compile_options flag to increase it.

Possible another PR then? Seems like this would be more involved.

**Comment by TedThemistokleous** [comment]
Total time:   3.37061ms ->   3.33602ms  with this change set off develop

**Comment by TedThemistokleous** [comment]
Closing this out as I've moved the changes for a matcher to be shared between other operations.

**Pull request closed without merging**